cardano-api/test/Test/Cardano/Api/Gen.hs

```diff
@@ -83,6 +136,54 @@ genSigningKeyShelley = do
     seedSize :: Int
     seedSize = fromIntegral (seedSizeDSIGN (Proxy :: Proxy (DSIGN TPraosStandardCrypto)))
 
+genShelleyStakePoolRegistrationCertificate :: Gen Certificate
+genShelleyStakePoolRegistrationCertificate =
+  shelleyRegisterStakePool
+    <$> genVerificationKeyHashStakePoolShelley
+    <*> genVRFVerificationKeyHashShelley
+    <*> (Coin <$> Gen.integral (Range.linear 0 10000000000))
+    <*> (Coin <$> Gen.integral (Range.linear 0 10000000000))
+    <*> genStakePoolMarginShelley
+    <*> genRewardAccountShelley
+    <*> genStakePoolOwnersShelley
+    <*> Gen.list (Range.linear 1 5) genStakePoolRelayShelley
+    <*> (Just <$> genPoolMetaDataShelley)
```

Comment:
If that genuinely is optional, you should use `Gen.maybe` which is `Nothing` about 30% of the time.